src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -328,7 +328,7 @@ public String formatHex(byte[] bytes, int index, int length) {
         // Format efficiently if possible
         String s = formatOptDelimiter(bytes, index, length);
         if (s == null) {
-            StringBuilder sb = new StringBuilder(bytes.length *
+            StringBuilder sb = new StringBuilder(length *
                     (delimiter.length() + prefix.length() + suffix.length()) - delimiter.length());
```

Comment:
should that be:
```
length * (delimiter.length() + prefix.length() + suffix.length() + 2) - delimiter.length());
```
It seems to me that otherwise, if you have no delimiter, no prefix, and no suffix, then your computed size will be 0, which seems wrong.

Reply:
Agreed, will fix.